backend/src/utils.rs

```diff
@@ -92,3 +92,107 @@ pub async fn get_deployments(env_vars: &EnvVars, username: &str) -> Res<Vec<Depl
 
     Ok(deployments)
 }
+
+/// Check if a user has permission to access a project
+pub async fn check_access(env_vars: &EnvVars, username: &str, project_name: &str) -> Res<()> {
+    let deployments_dir = &env_vars.deployments_dir;
+
+    let client = reqwest::Client::new();
+    let repo_path = format!("{}/{}", deployments_dir.display(), project_name);
+    let repo = Repository::open(repo_path)?;
+    let repo_url = repo
+        .find_remote("origin")?
+        .url()
+        .ok_or(anyhow!(
+            "Error: Origin remote URL not found for repo {project_name}."
+        ))?
+        .to_string();
+    let parsed_url = Url::from_str(&repo_url)?;
+    let mut url_paths = parsed_url
+        .path_segments()
+        .ok_or(anyhow!("Error parsing repository remote URL."))?;
+    let repo_owner = url_paths
+        .next()
+        .ok_or(anyhow!(
+            "Error parsing repository remote URL: Repo owner not found."
+        ))?
+        .to_string();
+    let repo_name = url_paths
+        .next()
+        .ok_or(anyhow!(
+            "Error parsing repository remote URL: Repo name not found."
+        ))?
+        .to_string();   
+    if repo_owner == env_vars.gh_org_name {
+        let collab_role = github::get_collaborator_role(
+            &client,
+            &env_vars.gh_org_admin_token,
+            &repo_owner,
+            &repo_name,
+            username,
+        )
+        .await?;
+
+        // `None` means the user is not a collaborator
+        if let Some(role) = collab_role.as_deref()
+            && (role == "maintain" || role == "admin")
+        {
+            return Ok(());
+        }
+    }
+    Err(anyhow!("User does not have permission to access this project."))
+}
+
+#[derive(Deserialize, Serialize)]
+/// Settings for a project
+pub struct ProjectSettings {
+    /// Subdirectory which is deployed (relative to the project root)
+    pub deploy_dir: String,
+}
+
+/// Get the project settings (stored in .maint on the top level of the project directory)
+pub async fn get_project_settings(env_vars: &EnvVars, project_name: &str) -> Res<ProjectSettings> {
+    let maint_file_path = format!(
+        "{}/{}/.maint",
+        env_vars.deployments_dir.display(),
+        project_name
+    );
+
+    if let Ok(maint_file_contents) = fs::read_to_string(maint_file_path).await {
+        Ok(ProjectSettings { deploy_dir: maint_file_contents.trim().into() })
+    } else {
+        Ok(ProjectSettings { deploy_dir: ".".into() } )
+    }
+}
+
+#[derive(Deserialize, Serialize)]
+/// A single environment variable for a project
+pub struct ProjectEnvVar {
```

Comment:
Can we just output a key-value JSON instead? (See below) dotenvy returns an iterator which can probably be serialized by serde_json.